### ISSUE TITLE
Downloading “Playing with Lego's” only downloads some most recent threadmarks

### ISSUE THREAD

**Issue by ZyX-I** [JimmXinu/FanFicFare#279]
When trying to use FanFicFare on [Playing with Lego's](https://forums.spacebattles.com/threads/playing-with-legos-worm-supcom.377328/) it downloads only ten least recent threadmarks (X..Afterword), additionally arranging them in reverse order. Command used

    fanficfare -b 0 -p 'https://forums.spacebattles.com/threads/playing-with-legos-worm-supcom.377328/'

(same result when `-b 0` omitted which is how I normally download).

Using FanFicFare 2.23.0 installed with `pip`, Python 2.7.14-r1 from main portage tree (Gentoo, `-r1` is build script revision), other PyPI packages: html2text 2018.1.9, chardet 3.0.4, beautifulsoup4 4.6.0, html5lib 1.0.1, six 1.11.0, webencodings 0.5.1.

**Comment by ZyX-I** [comment]
Based on what I see, this is exactly what you get if you go for `https://forums.spacebattles.com/threads/playing-with-legos-worm-supcom.377328/threadmarks.rss?category_id=1`. Parsing the HTML variant (replace `.rss` with nothing) should be fine, there also is all necessary information in `data-*` properties of `<li` tags.

**Comment by ZyX-I** [comment]
The following patch seems to work fine for me:

```Patch
--- /home/zyx/.local/lib64/python2.7/site-packages/fanficfare/adapters/base_xenforoforum_adapter.py.bak	2018-03-21 11:42:52.181509527 +0300
+++ /home/zyx/.local/lib64/python2.7/site-packages/fanficfare/adapters/base_xenforoforum_adapter.py	2018-03-21 11:52:25.963983127 +0300
@@ -20,7 +20,7 @@
 logger = logging.getLogger(__name__)
 import re
 import urllib2
-from xml.dom.minidom import parseString
+import datetime
 
 from ..htmlcleanup import stripHTML
 from .. import exceptions as exceptions
@@ -241,10 +241,6 @@
         threadmarks=[]
         tmcat_num=None
 
-        # convenience method.
-        def xml_tag_string(dom,tag):
-            return dom.getElementsByTagName(tag)[0].firstChild.data.encode("utf-8").decode('utf-8')
-
         for threadmarksa in threadmarksas:
             tmcat_num = threadmarksa['href'].split('category_id=')[1]
             # get from earlier <a> now.
@@ -254,16 +250,16 @@
             ## minimum_threadmarks below.
             if tmcat_name in self.getConfigList('skip_threadmarks_categories'):
                 continue
-            threadmark_rss_dom = parseString(self._fetchUrl(self.getURLPrefix()+'/'+threadmarksa['href'].replace('threadmarks','threadmarks.rss')).encode('utf-8'))
-            # print threadmark_rss_dom.toxml(encoding='utf-8')
-
-            for tmcat_index, item in enumerate(threadmark_rss_dom.getElementsByTagName("item")):
-                title = xml_tag_string(item,"title")
-                url = xml_tag_string(item,"link")
-                author = xml_tag_string(item,"dc:creator")
-                date = xml_tag_string(item,"pubDate")
-                ## Fri, 23 Jun 2017 16:52:57 +0000
-                date = makeDate(date[5:-6], '%d %b %Y %H:%M:%S') # toss day-of-week and TZ--locales.
+            threadmark_dom = self.make_soup(self._fetchUrl(self.getURLPrefix()+'/'+threadmarksa['href']).encode('utf-8'))
+            tmd_div = threadmark_dom.find('div', {'class': 'threadmarkList'})
+            tmd_ol = tmd_div.find('ol')
+
+            for tmcat_index, item in enumerate(tmd_ol.find_all('li')):
+                item_a = item.find('a')
+                title = stripHTML(item_a)
+                url = item_a['href'].encode('ascii')
+                author = item['data-content-author'].encode('utf-8')
+                date = datetime.datetime.fromtimestamp(int(item['data-content-date']))
                 threadmarks.append({"tmcat_name":tmcat_name,"tmcat_num":tmcat_num,"tmcat_index":tmcat_index,"title":title,"url":url,"date":date,"author":author})
         return threadmarks
 
```

I am not sure how universal it is though.

**Comment by jeancinq** [comment]
This problem is not specific to "playing with legos".
using calibre fanficfare plugin, it occurred to me with multiple threads on spacebattles and on sufficient velocity, but not on questionable questing.
version utilized:  2.23.0
options on the personal.ini
`[base_xenforoforum]
always_reload_first_chapter:true
description_limit:500000
normalize_text_links:true
internalize_text_links:true
continue_on_chapter_error:true
`

stories where I encountered this problem:
[the wolf time sv](https://forums.sufficientvelocity.com/threads/the-wolf-time-worm-au.17330/)
[miracles of ancient wonder sv](https://forums.sufficientvelocity.com/threads/miracles-of-ancient-wonder-rwby-exalted.28772/)
[limit theory sv](https://forums.sufficientvelocity.com/threads/limit-theory-pa-multicross-si.26647/)
[reverse engineering for fun and profit sv](https://forums.sufficientvelocity.com/threads/reverse-engineering-for-fun-and-profit-planetary-annihilation-multicross-si.24514/)
[fate denied sv](https://forums.sufficientvelocity.com/threads/fate-denied-a-star-wars-si.36312/)
[don't be the tsundere sv](https://forums.sufficientvelocity.com/threads/don't-be-the-tsundere-fate-stay-night-multicross-si.27312/)
[luv and hate sb](https://forums.spacebattles.com/threads/luv-and-hate-supcom-muv-luv-alternative-story-only-thread.281928/)

**Comment by JimmXinu** [comment]
Duplicate of #278.  Already fixed in ef0815e

**Comment by JimmXinu** [comment]
FYI: Uploaded a test CLI version.  You can install it with:

`pip install --extra-index-url https://testpypi.python.org/pypi --upgrade FanFicFare`
